target_embed_metal_shader_libraries invalid DEPENDS parameter

> this line causes the following error:
> 
> 
> 
> ```
> 
> CMake Error at cmake/MetalShaderSupport.cmake:46 (add_custom_command):
> 
>   The following keywords are not supported when using
> 
>   add_custom_command(TARGET): DEPENDS
> 
> Call Stack (most recent call first):
> 
>   CMakeLists.txt:99 (target_embed_metal_shader_libraries)
> 
> ```
> 
> 
> 
> according to cmake document 
> 
> 
> 
> https://cmake.org/cmake/help/latest/command/add_custom_command.html
> 
> 
> 
> DEPENDS is unsupported.  

 _Originally posted by @shi-yan in [6594c42](https://github.com/dpogue/CMake-MetalShaderSupport/commit/6594c420e2092f26fa050d897d69b5030e32f83d#r149884032)_

Since we can't use `DEPENDS` here, we should add the shaderlib as a dependency to the target manually.

--- a/cmake/MetalShaderSupport.cmake
+++ b/cmake/MetalShaderSupport.cmake
@@ -42,8 +42,8 @@ function(target_embed_metal_shader_libraries TARGET)
         )
     else()
         foreach(SHADERLIB IN LISTS _temsl_UNPARSED_ARGUMENTS)
+            add_dependencies(${TARGET} ${SHADERLIB})
             add_custom_command(TARGET ${TARGET} POST_BUILD
-                DEPENDS ${SHADERLIB}
                 COMMAND ${CMAKE_COMMAND} -E copy "$<TARGET_FILE:${SHADERLIB}>" "$<TARGET_BUNDLE_CONTENT_DIR:${TARGET}>/Resources/$<TARGET_FILE_NAME:${SHADERLIB}>"
                 VERBATIM
             )

